### ISSUE TITLE
Port availability much slower than Docker desktop on ARM macOS

### ISSUE THREAD

**Issue by yogurtearl** [rancher-sandbox/rancher-desktop#3141]
### Actual Behavior

It takes about 3 seconds for the ports in a postgres container to be made available when using Rancher Desktop.

See here:
https://github.com/testcontainers/testcontainers-java/issues/5986#issuecomment-1275496406

If I sleep for 4 seconds in the test, it will work. 

### Steps to Reproduce

Use `testcontainer-java` to load postgres arm image on macOS 12.6+ 

Test will fail with Rancher Desktop and success with Docker Desktop because Rancher Desktop take 1000x longer to expose the port.  `3000ms` on Rancher Desktop vs `3ms` on Docker Desktop

https://github.com/testcontainers/testcontainers-java/issues/5986#issuecomment-1275496406 

### Result

Test failures because Rancher Desktop took too long to expose the port. 

### Expected Behavior

Rancher Desktop should have similar performance characteristics as Docker Desktop

### Rancher Desktop Version

1.6.0

### Rancher Desktop K8s Version

none

### Which container engine are you using?

moby (docker cli)

### What operating system are you using?

macOS

### Operating System / Build Version

macOS 12.6

### What CPU architecture are you using?

arm64 (Apple Silicon)

**Comment by jandubois** [comment]
> It takes about 3 seconds for the ports in a postgres container to be made available when using Rancher Desktop.

Yes, this is unfortunately expected right now: Lima (the tool running the virtual machine and setting up port forwarding) is parsing `/proc/net/tcp` every 3 seconds to determine if any ports need to be opened or closed on the host: https://github.com/lima-vm/lima/blob/master/cmd/lima-guestagent/daemon_linux.go#L23

We have some code to watch for events from containerd, dockerd, and kubernetes, but it is currently only used in the Windows version. We are planning to merge it into Lima in the future, so macOS and Linux can also stop polling.

This issue will be closed once the code has been merged into Lima, and Rancher Desktop has been updated to the new Lima version afterwards.

**Comment by ArloL** [comment]
Is there an issue to follow in the lima project, @jandubois? I tried searching for `port` and `poll`.

**Comment by muawiakh** [comment]
Is there any update on this issue? I think we are facing a similar issue.

**Comment by daveisagit** [comment]
I'm using Rancher Desktop 1.22.0 and experiencing the same issue in that tests work if I put a `sleep(>2)` in my app start up. Is this still a required workaround for now?
